src/textures/CompressedArrayTexture.js

```diff
@@ -11,6 +11,20 @@ class CompressedArrayTexture extends CompressedTexture {
 		this.image.depth = depth;
 		this.wrapR = ClampToEdgeWrapping;
 
+		this.dirtyLayers = new Set();
+
+	}
+
+	addDirtyLayer( layerIndex ) {
+
+		this.dirtyLayers.add( layerIndex );
+
+	}
+
+	clearDirtyLayers() {
+
+		this.dirtyLayers.clear();
+
```

Comment:
Would it make sense to add the same new API to `DataArrayTexture` as well?

Reply:
Yes, as mentioned in my previous comment this would be definitely useful!

> This enhancement appears to also be beneficial for non-compressed array textures (DataArrayTexture). Consequently, we should consider updating texSubImage3D to reflect this change.

Reply:
One of the reasons I originally held off on doing this was the ambiguity around how many values in the data array encode a single texel

> The interpretation of the data depends on type and format: If the type is THREE.UnsignedByteType, a Uint8Array will be useful for addressing the texel data. If the format is THREE.RGBAFormat, data needs four values for one texel; Red, Green, Blue and Alpha (typically the opacity).
For the packed types, THREE.UnsignedShort4444Type and THREE.UnsignedShort5551Type all color components of one texel can be addressed as bitfields within an integer element of a Uint16Array.
In order to use the types THREE.FloatType and THREE.HalfFloatType, the WebGL implementation must support the respective extensions OES_texture_float and OES_texture_half_float. In order to use THREE.LinearFilter for component-wise, bilinear interpolation of the texels based on these types, the WebGL extensions OES_texture_float_linear or OES_texture_half_float_linear must also be present.

https://threejs.org/docs/#api/en/textures/DataArrayTexture

Is there existing code in ThreeJS I could reference which would describe for a given color space, how many values in the data array are used for a single texel? Otherwise, it would be challenging to correctly splice the texture array's layers.

Reply:
I'd say something like:
```js
let formatSize = 1

if ( glFormat === gl.RG ) formatSize = 2
if ( glFormat === gl.RGB ) formatSize = 3
if ( glFormat === gl.RGBA ) formatSize = 4
```
should do the trick? I don't think the type matter, only the format of the texture if I understand correctly the issue.

Also we could try to do this in another PR if you prefer.

Reply:
Made this change, that said the document is extremely sparse on how to correctly determine texel size. I supported the well documented formats and types and the remaining ones will throw a runtime error _only_ when using this specific feature. How do we feel about having any future developers who want to support additional formats/types contribute the texel calculation and we land this with the defaults it currently has?